aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionOperation.java

```diff
@@ -95,6 +96,9 @@ public synchronized void start() {
     public synchronized void cancel() {
         if (subscriptionId != null && !canceled) {
             canceled = true;
+            if (subscriptionFuture.cancel(true)) {
```

Comment:
Possible NPE since subscriptionFuture could be null?

Reply:
Yup...accounting for that now.